src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -311,21 +369,62 @@ public InvocationHandler(Type[] parameterTypes, Action<object[]> handler)
             }
         }
 
-        private struct InvocationRequest
+        private class InvocationRequest : IDisposable
         {
+            private readonly TaskCompletionSource<object> _completionSource = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+            private readonly CancellationTokenRegistration _cancellationTokenRegistration;
+            private object _result;
+            private object _lock = new object();
```

Comment:
You're right, it is now. This went through a few different iterations and the TCS was my last addition.